core/src/CMakeLists.txt

```diff
@@ -19,4 +19,8 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/CabanaCore_config.hpp DESTINATION ${CM
 add_library(cabanacore ${SOURCES} ${SOURCES} ${SOURCES_IMPL})
 target_include_directories(cabanacore PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
 target_link_libraries(cabanacore Kokkos::kokkos)
+if(Cabana_ENABLE_MPI)
+  find_package(MPI REQUIRED)
```

Comment:
Not sure I'd put the `find_package(MPI)` here.  You might want to do that in the root directory `CMakeLists.txt` so you can register test with `${MPIEXEC} ${MPIEXEC_NUMPROC_FLAG} ...`

Also do not forget the `if (NOT MPI_CXX_FOUND) message(FATAL_ERROR ...`

Reply:
The former is a good point as tests might not live in a subfolder!

The latter, I was thinking `REQUIRED` will check for all enabled languages, incl. `CXX`.

Reply:
@junghans Do you want to make these changes or should I?

Reply:
@sslattery I did the former, I don't think the latter is necessary.

Reply:
> The latter, I was thinking REQUIRED will check for all enabled languages, incl. CXX.

Oh that's right